docs/specifications/cva6_requirement_specification.adoc

```diff
@@ -0,0 +1,391 @@
+
+=== CSR
+
+There are no requirements related to CSR as they derive from other requirements, such as PVL-10, PVL-60… Details of CSRs will be available in the user’s manual.
+
+=== Performance counters
+
+Performance counters are important features for safety-critical applications.
+|===
+
+|HPM&#8209;10|CVA6 shall implement the 64-bit `mcycle` and `minstret` standard performance counters (including their upper 32 bits counterparts `mcycleh` and `minstreth` in CV32A6) as per [RVpriv].
+|HPM&#8209;20|CVA6 shall implement as an *option* six generic 64-bit performance counters located in `hpmcounter3` to `hpmcounter8` (including their upper 32 bits counterparts in CV32A6: `hpmcounter3h` to `hpmcounter8h`).
+|HPM&#8209;30|Each of the six generic performance counters shall be able to count events from one of these sources: +
+1. L1 I-Cache misses +
+2. L1 D-Cache misses +
+3. ITLB misses +
+4. DTLB misses +
+5. Load accesses + 
+6. Store accesses +
+7. Exceptions +
+8. Exception handler returns +
+9. Branch instructions +
+10. Branch mispredicts + 
+11. Branch exceptions + 
+12. Call +
+13. Return +
+14. MSB Full +
+15. Instruction fetch Empty +
+16. L1 I-Cache accesses + 
+17. L1 D-Cache accesses +
+18. L1$ line invalidation +
+19. I-TLB flush +
+20. Integer instructions + 
+21. Floating point instructions +
+22. Pipeline bubbles
+|HPM&#8209;40|The source of events counted by the six generic performance counters shall be selected by the `mhpmevent3` to `mhpmevent8` CSRs.
+|HPM&#8209;50|CVA6 shall allow the supervisor access of performance counters through enabling of `mcounteren` CSR.
+|HPM&#8209;60|CVA6 shall allow the user access of performance counters through enabling of `scounteren` CSR.
+|HPM&#8209;70|CVA6 shall implement the `mcountinhibit` counter-inhibit register.
+|HPM&#8209;80|CVA6 shall implement the read-only `cycle`, `instret`, `hpmcounter3` to `hpmcounter8` access to counters (and their upper 32-bit counterparts in CV32A6).
+|===
+
+The user’s manual will detail the list of counters, events and related controls.
+
+=== Cache requirements  
+
+Caches increase the performance of the processor with regard to memory accesses. Most of their added value for the IP is specified through performance requirements in another section. Here below are specific requirements for these caches.
+
+The project would like to adopt the recently ratified [RVcmo] specification. The analysis yet needs to be performed and will likely lead to an evolution of this specification.
+
+==== L1 write-through data cache
+
+In the requirements below, L1WTD refers to the L1 write-through data cache that is part of the CVA6.
+
+The first two requirements express the write-through feature. Some requirements are useful for security- and safety-critical applications where a high level of timing predictability is needed.
+
+|===
+
+|L1W&#8209;10|L1WTD shall reflect all write accesses (stores) by the CVA6 core to the external memory after an upper-bounded number of cycles.
```

Comment:
is that a configurable or fixed upper-bound? Also  I think you mean "within" an upper-bounded number of cycles, not "after".

Reply:
This is not configurable. It just means that L1WTD shall not wait forever to reflect the access write; otherwise it would be a write-back cache. Any better wording suggestion is appreciated.

Reply:
L1WTD shall reflect all write accesses (stores) by the CVA6 core to the external memory within an upper-bounded number of cycles. The upper-bound is fixed but not specified here.

Reply:
Implemented.